src/ui/public/dialogues/confirm_dialogue.js

```diff
@@ -0,0 +1,63 @@
+import uiModules from 'ui/modules';
+import confirmDialogueTemplate from './confirm_dialogue.html';
+
+const module = uiModules.get('kibana');
+
+module.factory('showConfirmDialogue', function (ModalDialogue, $q, $compile, $rootScope) {
+  /**
+   * Shows a modal confirmation dialog with the given message.
+   *
+   * @param {String} message
+   * @param {String} yesButtonText - Defaults to Okay, supply a custom string to override.
+   * @param {String} noButtonText - Defaults to Cancel, supply a custom string to override.
+   * @return {Promise<boolean>} Returns a promise that will be resolved to true if the user
+   * clicks the yes/okay button and rejected if the user clicks the no/cancel button.
+   */
+  return function showConfirmDialogue(message, yesButtonText = 'Okay', noButtonText = 'Cancel') {
+    const deferred = $q.defer();
+
+    const dialogueScope = $rootScope.$new();
+
+    let modalDialogue = undefined;
+
+    dialogueScope.onYes = () => {
+      dialogueScope.$destroy();
+      modalDialogue.destroy();
+      deferred.resolve(true);
```

Comment:
I did that to match the prior functionality since it was tested, and I wanted to keep it how it was before.  Otherwise I have to check all the call sites and make sure all usages of safeConfirm aren't relying on true/false.  I thought it was silly to, to pass this, just worried about breaking something.  If you feel strongly about it though, I can do it.